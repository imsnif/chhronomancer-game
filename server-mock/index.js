import playerStore from '../stores/player'
import timelineStore from '../stores/timeline'
import powerStore from '../stores/power'
import statsStore from '../stores/stats'

const timelines = [
  {name: 'Timeline 1', type: 'steal', players: []},
  {name: 'Timeline 2', type: 'assist', players: []},
  {name: 'Timeline 3', type: 'prevent', players: []},
  {name: 'Timeline 4', type: 'reset', players: []}
]

const players = [
  {
    id: 1,
    name: 'Aram',
    items: [
      {name: 'assist', source: false},
      {name: 'prevent', source: 'Timeline 3'},
      {name: 'prevent', source: 'Timeline 3'},
      {name: 'reset', source: 'Timeline 4'},
      {name: 'reset', source: 'Timeline 4'},
      {name: 'steal', source: 'Timeline 1'},
      {name: 'lock', source: false},
      {name: 'unlock', source: false}
    ],
    actions: 10,
    image: require('../assets/placeholders/green_me_s.png')
  },
  {
    id: 2,
    name: 'Tom',
    items: [
      {name: 'assist', source: 'Timeline 2'},
      {name: 'prevent', source: 'Timeline 3'},
      {name: 'reset', source: 'Timeline 3'},
      {name: 'steal', source: 'Timeline 1'},
      {name: 'lock', source: 'Timeline 1'},
      {name: 'unlock', source: false}
    ],
    actions: 10,
    image: require('../assets/placeholders/green_tom_s.png')
  },
  {
    id: 3,
    name: 'Gondollieri',
    items: [
      {name: 'assist', source: 'Timeline 2'},
      {name: 'prevent', source: 'Timeline 3'},
      {name: 'reset', source: 'Timeline 3'},
      {name: 'steal', source: 'Timeline 1'},
      {name: 'lock', source: 'Timeline 3'},
      {name: 'unlock', source: 'Timeline 1'}
    ],
    actions: 10,
    image: require('../assets/placeholders/green_gondollieri_s.png')
  },
  {
    id: 4,
    name: 'Simmons',
    items: [
      {name: 'assist', source: 'Timeline 2'},
      {name: 'prevent', source: false},
      {name: 'reset', source: 'Timeline 3'},
      {name: 'steal', source: 'Timeline 1'},
      {name: 'lock', source: 'Timeline 2'},
      {name: 'unlock', source: 'Timeline 2'}
    ],
    actions: 10,
    image: require('../assets/placeholders/green_simmons_s.png')
  }
]

const powers = [
  {
    playerId: 1,
    timelineName: 'Timeline 1',
    name: 'Resetting',
    startTime: Date.now(),
    endTime: Date.now() + 100000,
    target: {
      type: 'timeline',
      name: 'Timeline 1'
    },
    allies: [
      {id: 1, score: 8},
      {id: 2, score: 5}
    ],
    enemies: [
      {id: 3, score: 1},
      {id: 4, score: 9}
    ]
  },
  {
    playerId: 1,
    timelineName: 'Timeline 2',
    name: 'Locking',
    startTime: Date.now(),
    endTime: Date.now() + 400000,
    target: {
      type: 'timeline',
      name: 'Timeline 2'
    },
    allies: [
      {id: 1, score: 8},
      {id: 2, score: 5}
    ],
    enemies: [
      {id: 3, score: 1},
      {id: 4, score: 9}
    ]
  },
  {
    playerId: 2,
    timelineName: 'Timeline 1',
    name: 'Resetting',
    startTime: Date.now(),
    endTime: Date.now() + 1000000,
    target: {
      type: 'timeline',
      name: 'Timeline 1'
    },
    allies: [
      {id: 2, score: 1},
      {id: 1, score: 5}
    ],
    enemies: [
      {id: 3, score: 1},
      {id: 4, score: 2}
    ]
  },
  {
    playerId: 3,
    timelineName: 'Timeline 1',
    name: 'Combining',
    startTime: Date.now(),
    endTime: Date.now() + 2000000,
    target: {
      type: 'timeline',
      name: 'Timeline 1'
    },
    allies: [
      {id: 3, score: 1},
      {id: 1, score: 2}
    ],
    enemies: [
      {id: 2, score: 2},
      {id: 4, score: 5}
    ]
  },
  {
    playerId: 4,
    timelineName: 'Timeline 1',
    name: 'Locking',
    startTime: Date.now(),
    endTime: Date.now() + 100000,
    target: {
      type: 'timeline',
      name: 'Timeline 1'
    },
    allies: [
      {id: 4, score: 7},
      {id: 2, score: 2}
    ],
    enemies: [
      {id: 2, score: 2},
      {id: 1, score: 5}
    ]
  }
]

export default function mockServer () {
  timelines.forEach((t) => timelineStore.addTimeline(t))
  players.forEach(p => {
    playerStore.addPlayer(p)
  })
  statsStore.changePlayerId(1)
  timelineStore.addPlayer('Timeline 1', 1)
  timelineStore.addPlayer('Timeline 1', 2)
  timelineStore.addPlayer('Timeline 2', 2)
  timelineStore.addPlayer('Timeline 3', 2)
  timelineStore.addPlayer('Timeline 1', 3)
  timelineStore.addPlayer('Timeline 2', 3)
  timelineStore.addPlayer('Timeline 3', 3)
  timelineStore.addPlayer('Timeline 1', 4)
  timelineStore.addPlayer('Timeline 3', 4)
  powers.forEach(p => powerStore.addPower(p))
  timelineStore.changeTimelineLock('Timeline 1', true)
}
