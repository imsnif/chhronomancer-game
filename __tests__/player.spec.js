import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import '../server-mock'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))
jest.useFakeTimers()

test('Player (no items or power) => renders correctly', () => {
  const Player = require('../components/player').default
  const tree = renderer.create(
    <Player id={1} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Player (with items and no power) => renders correctly', () => {
  const Player = require('../components/player').default
  const tree = renderer.create(
    <Player id={2} timelineName='Timeline 2' />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Player (with items and power) => renders correctly', () => {
  const Player = require('../components/player').default
  const tree = renderer.create(
    <Player id={2} timelineName='Timeline 1' />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
