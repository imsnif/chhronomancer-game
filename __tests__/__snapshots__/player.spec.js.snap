// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player (default props) => renders correctly 1`] = `
<View
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#00ff00",
        "borderColor": "#00ff00",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 1,
        "flexDirection": "column",
        "height": 100,
        "margin": 5,
        "marginBottom": 15,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={undefined}
        style={
          Object {
            "backgroundColor": "black",
            "borderRadius": 0,
            "flex": 1,
            "height": 21,
            "margin": 2,
            "marginBottom": 1,
            "width": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "flex": 9,
            "fontFamily": "px_sans_nouveaux",
            "paddingRight": 30,
            "textAlign": "center",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "black",
          "borderRadius": 0,
          "borderWidth": 0.5,
          "flex": 3,
          "flexDirection": "column",
          "margin": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "red",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "green",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "blue",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "purple",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "brown",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "yellow",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Player (with activePower and items) => renders correctly 1`] = `
<View
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#00ff00",
        "borderColor": "#00ff00",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 1,
        "flexDirection": "column",
        "height": 100,
        "margin": 5,
        "marginBottom": 15,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={undefined}
        style={
          Object {
            "backgroundColor": "black",
            "borderRadius": 0,
            "flex": 1,
            "height": 21,
            "margin": 2,
            "marginBottom": 1,
            "width": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "flex": 9,
            "fontFamily": "px_sans_nouveaux",
            "paddingRight": 30,
            "textAlign": "center",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "black",
          "borderRadius": 0,
          "borderWidth": 0.5,
          "flex": 3,
          "flexDirection": "column",
          "margin": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "black",
            "flex": 1,
            "padding": 5,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#00ff00",
              "borderRadius": 0,
              "flex": 1,
              "padding": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "black",
                "borderRadius": 0,
                "flex": 1,
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgba(2, 255, 255, 1)",
                  "borderRadius": 0,
                  "flex": 1,
                  "height": "100%",
                  "position": "absolute",
                  "width": "2%",
                }
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "transparent",
                  "elevation": 6,
                  "flexDirection": "row",
                  "height": "100%",
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#00ff00",
                    "flex": 2,
                    "fontFamily": "px_sans_nouveaux",
                    "marginLeft": 4,
                  }
                }
              >
                PowerName
                ...
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#00ff00",
                    "flex": 2,
                    "fontFamily": "px_sans_nouveaux",
                    "marginTop": 1,
                  }
                }
              >
                00:00:09
              </Text>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "marginTop": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#00ff00",
                      "fontFamily": "px_sans_nouveaux",
                    }
                  }
                >
                  3
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#00ff00",
                      "fontFamily": "px_sans_nouveaux",
                    }
                  }
                >
                  /
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#00ff00",
                      "fontFamily": "px_sans_nouveaux",
                    }
                  }
                >
                  4
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "red",
                  "borderColor": "red",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "green",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "blue",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "purple",
                  "borderColor": "purple",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "brown",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "yellow",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Player (with activePower) => renders correctly 1`] = `
<View
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#00ff00",
        "borderColor": "#00ff00",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 1,
        "flexDirection": "column",
        "height": 100,
        "margin": 5,
        "marginBottom": 15,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={undefined}
        style={
          Object {
            "backgroundColor": "black",
            "borderRadius": 0,
            "flex": 1,
            "height": 21,
            "margin": 2,
            "marginBottom": 1,
            "width": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "flex": 9,
            "fontFamily": "px_sans_nouveaux",
            "paddingRight": 30,
            "textAlign": "center",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "black",
          "borderRadius": 0,
          "borderWidth": 0.5,
          "flex": 3,
          "flexDirection": "column",
          "margin": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "black",
            "flex": 1,
            "padding": 5,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#00ff00",
              "borderRadius": 0,
              "flex": 1,
              "padding": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "black",
                "borderRadius": 0,
                "flex": 1,
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgba(2, 255, 255, 1)",
                  "borderRadius": 0,
                  "flex": 1,
                  "height": "100%",
                  "position": "absolute",
                  "width": "2%",
                }
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "transparent",
                  "elevation": 6,
                  "flexDirection": "row",
                  "height": "100%",
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#00ff00",
                    "flex": 2,
                    "fontFamily": "px_sans_nouveaux",
                    "marginLeft": 4,
                  }
                }
              >
                PowerName
                ...
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#00ff00",
                    "flex": 2,
                    "fontFamily": "px_sans_nouveaux",
                    "marginTop": 1,
                  }
                }
              >
                00:00:09
              </Text>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "marginTop": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#00ff00",
                      "fontFamily": "px_sans_nouveaux",
                    }
                  }
                >
                  3
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#00ff00",
                      "fontFamily": "px_sans_nouveaux",
                    }
                  }
                >
                  /
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#00ff00",
                      "fontFamily": "px_sans_nouveaux",
                    }
                  }
                >
                  4
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "red",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "green",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "blue",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "purple",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "brown",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "yellow",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Player (with items) => renders correctly 1`] = `
<View
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#00ff00",
        "borderColor": "#00ff00",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 1,
        "flexDirection": "column",
        "height": 100,
        "margin": 5,
        "marginBottom": 15,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={undefined}
        style={
          Object {
            "backgroundColor": "black",
            "borderRadius": 0,
            "flex": 1,
            "height": 21,
            "margin": 2,
            "marginBottom": 1,
            "width": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "flex": 9,
            "fontFamily": "px_sans_nouveaux",
            "paddingRight": 30,
            "textAlign": "center",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "black",
          "borderRadius": 0,
          "borderWidth": 0.5,
          "flex": 3,
          "flexDirection": "column",
          "margin": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "red",
                  "borderColor": "red",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "green",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "blue",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "purple",
                  "borderColor": "purple",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "brown",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderColor": "black",
                "borderRadius": 0,
                "borderWidth": 1,
                "height": 14,
                "margin": 4,
                "width": 14,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderColor": "yellow",
                  "borderWidth": 1,
                  "height": 8,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 8,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
